packages/hyperion-autologging/src/ALSurface.ts

```diff
@@ -180,46 +206,87 @@ function setupDomElementSurfaceAttribute(options: InitOptions): void {
+  const { flowletManager, domSurfaceAttributeName = AUTO_LOGGING_SURFACE, domRegionAttributeName = AUTO_LOGGING_REGION } = options;
   const { ReactModule } = options.react;
 
   ALIReactFlowlet.init(options);
 
   setupDomElementSurfaceAttribute(options);
   const SurfaceContext = ALSurfaceContext.init(options);
 
+  type SurfaceData = ALSurfaceContext.ALSurfaceContextFilledValue & {
+    domAttributeName: string;
+    domAttributeValue: string;
+  }
+
   const Surface: SurfaceComponent = props => {
-    const { __ext, flowlet, flowletManager } = props;
+    const { __ext, flowlet, proxiedContext } = props;
     if (__ext && __ext.flowlet !== flowlet) {
       __ext.flowlet = flowlet;
     }
 
-    const incomingSurfaceString = props.fullSurfaceString ?? '';
-    const isPassedSurface = incomingSurfaceString !== '';
-    let fullSurfaceString = incomingSurfaceString;
-    const { surface: parentSurface } = ALSurfaceContext.useALSurfaceContext();
-    if (!isPassedSurface) {
+    let fullSurfaceString: string;
+    let fullRegionString: string;
+    let domAttributeName: string;
+    let domAttributeValue: string;
+
+    const surfaceCtx = ALSurfaceContext.useALSurfaceContext();
+    const { surface: parentSurface, region: parentRegion } = surfaceCtx;
+
+    if (!proxiedContext) {
       const surface = flowlet.name;
-      fullSurfaceString = (parentSurface ?? '') + SURFACE_SEPARATOR + surface;
+      fullRegionString = (parentRegion ?? '') + SURFACE_SEPARATOR + surface;
+      const trackInteraction = props.capability == null || (props.capability & ALSurfaceCapability.TrackInteraction); // empty .capability field is default, means all enabled! 
+      if (!trackInteraction) {
+        fullSurfaceString = parentSurface ?? SURFACE_SEPARATOR;
+        domAttributeName = domRegionAttributeName;
+        domAttributeValue = fullRegionString;
+      } else {
+        fullSurfaceString = (parentSurface ?? '') + SURFACE_SEPARATOR + surface;
+        domAttributeName = domSurfaceAttributeName;
+        domAttributeValue = fullSurfaceString;
+      }
+    } else {
+      fullSurfaceString = proxiedContext.surface;
+      fullRegionString = proxiedContext.region;
+      domAttributeName = domSurfaceAttributeName
+      domAttributeValue = fullSurfaceString;
     }
 
-    if (options.channel && !isPassedSurface) {
+    const surfaceData: SurfaceData = {
+      surface: fullSurfaceString,
+      region: fullRegionString,
+      flowlet,
+      domAttributeName,
+      domAttributeValue,
+    };
+
+    if (options.channel && !proxiedContext) {
       const { channel } = options;
       // Emit surface mutation events on mount/unmount
+      const metadata = props.metadata ?? {}; // Note that we want the same object to be shared between events to share the changes.
+      metadata.original_flowlet = flowlet.getFullName();
+
+      metadata.surface_capability = surfaceCapabilityToString(props.capability);
```

Comment:
This is a actually a set, but it is a bit set. In future, when we add more options, once call pass multiple choices by simply ORing them. 
E.g. `capability = ALSurfaceCapability.TracInteraction | ALSurfaceCapability.TrackMutation`. That is super efficient and fast to handle. 
Is that what you were asking about? 
I could also choose an array of enums, but then it will be more verbose to pass, and slower to check.

Reply:
Ahh yeah. Got it, that's pretty cool!